main.go

```diff
@@ -348,6 +348,7 @@ func {{.FunctionPrefix}}FSByte(useLocal bool, name string) ([]byte, error) {
 		if err != nil {
 			return nil, err
 		}
+		defer f.Close()
```

Comment:
Defer is a slow operation, and so shouldn't be used if there's only one return path. There's no difference in the return values if ReadAll returns an error, so the extra logic you have above doesn't add any value.